### ISSUE TITLE
Use gpg (and potentially other) from user provided git installation

### ISSUE THREAD

**Issue by achobanov** [cmderdev/cmder#2577]
### Purpose of the issue
- [ ] Bug report (encountered problems/errors)
- [ ] Feature request (request for new functionality)
- [x] Question

### Version Information
Cmder: 1.3.18
ConEmu: 210304 preview

### Description of the issue
How can I configure Cmder to use the gpg.exe from my specified git installation? I've tried adding it's PATH, but it doesn't work. It seems reasonable to me that if I am allowed to use more up-to-date version of git, I can also use the latest version of it's vendors apps.

**Comment by 296951** [comment]
Cmder automatically searches for and uses the most current Git install that exists in the path.

If you:

- Install Git for Windows
- It is newer than Cmder Full 's embedded Git
- It is in the path.

Cmder will find and use it for `cmd.exe` based shells.

I believe you can avoid this auto detection process by setting the env variable in Windows advanced settings `GIT_INSTALL_ROOT=[Git for Windows Install Path]`

Example: `GIT_INSTALL_ROOT=C:\Program Files\Git`

**Comment by achobanov** [comment]
Yes, you are correct. But it only works for Git. Cmder indeed uses my version of git, but I am talking about the `gpg.exe `here. Whatever I tried to do it kept on sticking with the `gpg.exe` from the Cmder's vendored Git. Which potentially means that I am not using the latest version of encryption and that could be  a problem. 

Furthermore, how can I add another application to Cmder? I'm trying to install Istio right now. I have it extracted on my PC and I have a shortcut pointing towards the .exe file inside `cmder_cfg/bin`  directory. Nothing happens. I've also tried extracting the whole of Istio there and still `command not found`. Do I somehow misunderstand the config for Cmder, described by the Readme?

**Comment by daxgames** [comment]
By default Cmder adds `%cmder_root%\cmd` to the front of the path and `%cmder_root%\vendor\git-for-windows\mingw64\bin;%cmder_root%\vendor\git-for-windows\usr\bin;%cmder_root%\vendor\bin`.

Do you have a newer Git for WIndows installed or do you have a spearate install of `gpg` installed?

If you have a newer install of Git for Windows installed the autodetection built into Cmder has this same behavior and will use the newer `gpg` from the newer Git for Windows install. as shown below:.

![image](https://user-images.githubusercontent.com/7318053/127860939-d2e572fc-d8fd-40ad-8f73-226747d42df5.png)

Gpg location in Git for Windows is: `%GIT_INSTALL_ROOT%\usr\bin\gpg.exe`

If you have a newer standalone version of gpg installed and if the folder for gpg is in the path before launching Cmder your gpg should be used.

another option is to create an alias:

```
alias gpg="[path to gpg.exe]" $*
```

Regarding adding additional applications to Cmder.

You can use any installed app from Cmder by typing the fully qualified path to the file, creting an aliasas shown above, or adding the files parent folder to the path.  If you want to add the install of the app to Cmder so it is portable and thus follows Cmder if you copy it to another computer then you can put other PORTABLE apps in the `%cmder_root%\bin` folder.

The Cmder init script `init.bat`  has the ability to look for executable file paths multiple levels below `%cmder_root%\bin` by adding a command line switch to the task that starts Cmder but this slows down sessions start SIGNIFICANTLY so I do not recommend it. See: [Changing Cmder Default cmd.exe Shell Startup Behaviour Using Task Arguments](https://github.com/cmderdev/cmder#changing-cmder-default-cmdexe-shell-startup-behaviour-using-task-arguments)

See [Install VS Code in portable mode](https://github.com/cmderdev/cmder/wiki/Seamless-VS-Code-Integration#install-vs-code-in-portable-mode) for an example of adding a portable app to Cmder.

**Comment by daxgames** [comment]
@achobanov 

See [latest unreleased build](https://ci.appveyor.com/project/MartiUK/cmder/build/artifacts).  I think this fixes the reported issue.
